mcts4py/Solver.py

```diff
@@ -71,7 +71,7 @@ def calculate_uct_impl(self, parentN: TNode, n: TNode, reward: float, exploratio
         return reward/n + exploration_constant * sqrt(log(parentN)/n)
 
     def extract_optimal_action(self) -> Optional[TAction]:
-        return max(self.root().get_children(), key=lambda c: c.n)
+        return max(self.root().get_children(), key=lambda c: c.reward/c.n)
```

Comment:
This uses average reward instead of visit count, theoretically these are correlated so it shouldn't matter but this is an alternative.